src/AutoRest.CSharp/Mgmt/Decorator/TypeExtensions.cs

```diff
@@ -40,6 +40,20 @@ public static bool EqualsByName(this CSharpType left, CSharpType right)
             return true;
         }
 
+        public static bool EqualsBySystemType(this CSharpType? left, CSharpType? right)
+        {
+            if (ReferenceEquals(left, right))
+            {
+                return true;
+            }
+
+            if (left is null || right is null)
+            {
+                return false;
+            }
+            return left.GetType().Equals(right.GetType());
+        }
```

Comment:
is this method expected? 
Please note `left.GetType()` will always returns you `typeof(CSharpType)`
If you want to check if the underlying type of these two CSharpType instances, you should do:
`left.FrameworkType.Equals(right.FrameworkType)`
Also `CSharpType` class has an implementation of `Equals` to compare whether two types are the same type, and some other variant of this method such as `EqualsIgnoreNullable` etc.

What is the purpose of this method?

Reply:
`typeof(CSharpType)` just get the type info, not the other details, which is taken into consideration in `CSharpType.Equals` method. Found that [TypeExtension.EqualsByName](https://github.com/Azure/autorest.csharp/blob/63b56fd85c988fe795e1ed7829fbad1f09a35c18/src/AutoRest.CSharp/Mgmt/Decorator/TypeExtensions.cs#L26) serves the purpose, I just need to update it to nullable.